### PR TITLE
0.54.1 regression: add library() check for Fortran configure_file()

### PR DESCRIPTION
**Pull request by scivision**
Meson 0.54.1 has a regression noted in #7265. This additional test code reveals the issue, but does not fix the issue. This is an important use case that may only break on Windows, but it's good to add this test in general.

### CODE REVIEW

File: test cases/fortran/7 generated/meson.build [hunk 1]
```diff
@@ -6,6 +6,11 @@ project('generated', 'fortran')
 conf_data = configuration_data()
 conf_data.set('ONE', 1)
 conf_data.set('TWO', 2)
+conf_data.set('THREE', 3)
+
+outfile = configure_file(
+      input : 'mod3.fpp', output : 'mod3.f90', configuration : conf_data)
+three = library('mod3', meson.current_build_dir() / 'mod3.f90')
```
Comment (nirbheek) [inline test cases/fortran/7 generated/meson.build:13]: Why aren't you using `outfile` here directly instead of creating a string to the output?

**Comment by nirbheek** [comment]
@scivision poke :)

File: test cases/fortran/7 generated/meson.build (hunk 1 above)
Comment (nirbheek) [reply test cases/fortran/7 generated/meson.build:13]: I see, you want to test that an absolute path to a source file works correctly.

**Comment by nirbheek** [comment]
> The Cygwin failure should go away on a rebase but this one seems relevant:

Yes, that's what the test is checking for. The commit I just pushed should fix it.

**Comment by scivision** [comment]
Sorry for my delay. Yes I should have more simply done

```meson
three = library('mod3', sources: outfile)
```

it was just a simple oversight.

**Comment by nirbheek** [comment]
> it was just a simple oversight.

I think that would've used a relative path (and if it doesn't, it might've changed to use a relative path in the future), so creating the absolute path by hand is correct. I added a comment saying that when I added my commit :)

**Pull request merged**